pkg/api/handlers/compat/events.go

```diff
@@ -68,24 +77,52 @@ func GetEvents(w http.ResponseWriter, r *http.Request) {
 		}
 	}(r.Context().Done())
 
-	// Headers need to be written out before turning Writer() over to json encoder
-	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(http.StatusOK)
-	if flusher, ok := w.(http.Flusher); ok {
-		flusher.Flush()
-	}
+	// Write the events from the channel.
+	go func() {
+		first := true
+		var coder *jsoniter.Encoder
+		json := jsoniter.ConfigCompatibleWithStandardLibrary
 
-	json := jsoniter.ConfigCompatibleWithStandardLibrary
-	coder := json.NewEncoder(w)
-	coder.SetEscapeHTML(true)
+		writeHeader := func() {
+			// Headers need to be written out before turning
+			// Writer() over to json encoder
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			if flusher, ok := w.(http.Flusher); ok {
+				flusher.Flush()
+			}
 
-	for event := range eventChannel {
-		e := entities.ConvertToEntitiesEvent(*event)
-		if err := coder.Encode(e); err != nil {
-			logrus.Errorf("unable to write json: %q", err)
+			coder = json.NewEncoder(w)
+			coder.SetEscapeHTML(true)
 		}
-		if flusher, ok := w.(http.Flusher); ok {
-			flusher.Flush()
+
+		for event := range eventChannel {
```

Comment:
Do we have a guarantee this terminates in non-error conditions? I don't think that the channel is closed automatically when events run out, so we could block forever here.